### PR TITLE
Update Plone version in api backend folder to 5.2.6

### PR DESCRIPTION
**Pull request by fredvd**
And replace the kitconcept default README.rst buildout info with info on when to use this folder or preferably use docker to test Volto.

### CODE REVIEW

**Comment by fredvd** [comment]
This is the underlying issue, zope.security cannot be installed from wheel at the moment for aarch64, any issue with the build environment trips up a 'normal' install from source on M1. 

https://github.com/zopefoundation/zope.security/issues/78

**Review by tisto — APPROVED** [review]
@fredvd awesome! Thanks Fred!

**Review by nzambello — APPROVED** [review]

**Comment by fredvd** [comment]
This PR should work fine on X86 machines (win/lin/mac), but as more and more people use M1 macs, please hold merging this until I've figured how I install Plone 5.2.6 myself (or even install zope.security through pip) on an M1 with Monterey :-S

**Comment by nzambello** [comment]
> This PR should work fine on X86 machines (win/lin/mac), but as more and more people use M1 macs, please hold merging this until I've figured how I install Plone 5.2.6 myself (or even install zope.security through pip) on an M1 with Monterey :-S

@fredvd I had no issue installing Plone 5.2.6 on Python 3.8 with M1 on Monterey TBH, even with Plone 6 alpha.
I'm using `zope.security-5.1.1-py3.8-macosx-12-arm64`

**Comment by fredvd** [comment]
@nzambello interesting. I only have one machine where I can test M1, so I didn't exclude yet that it might be some very local edge case / mistake I made earlier. which python3.8 are you using and how is it installed? I'm using pyenv. And I updated to Monterey last week from big sur. My existing buildouts are still running but can't exclude yet it's not related.

buildout throws a traceback when trying to install the zope.security package where it complains when building a wheel that the file location already exists.  Didn't copy the exact error to my notebook. :-/

I had somewhat related issues last month where the environment of a subdependency install of an package is not correct, but it's difficult to debug/inspect because it is all temporary. But that one pulled in a wrong wheel from the local pip cache in the  dep build step. :-S

Travelling now, I spent some hours on this the weekend, will do some more debugging.

but if others see no issues then this PR can be merged and it a local issue on my machine 🥲

**Comment by nzambello** [comment]
@fredvd I use python@3.8 installed from brew and virtualenv for buildout envs.

**Comment by sneridagh** [comment]
@fredvd did you try again? any other insight?

@iFlameing @jackahl Could you please test on your M1s?

**Comment by fredvd** [comment]
> @fredvd did you try again? any other insight?
> @iFlameing @jackahl Could you please test on your M1s?

@sneridagh . Yes I have some insights. But i'm not there yet. You can safely merge this PR when it works for others because it's definitely a local edge case on my machine. 

But it's an edge case many more people will run into in the future if they use pyenv. 

I switched back to my good old Intel MBpro  to do the daily work and meanwhile debug/investigate the configuration on my M1 Mac Mini when I had some time left for the last 2 weeks.  The main culprit is a combination of me using pyenv for my Python 3 installs, attempting (and succeeding) in compiling python2 using pyenv with Rosetta in 2020 on Big Sur in x86 mode and Apple's Xcode / CommandLineTools (CLT)   MacOSX.SDK mess. 

I found this out by testing more after  @nzambello mentionned that Python3.8 from Homebrew worked fine for him. So I disabled pyenv and started testing purely with a Homebrew python3.

TL/DR:  Use Homebrew for now, because pyenv can become a minefield of differently picked up compiler/linker paths to different libraries between the Python3 executable and later Python modules that also need to link to the same or other system libraries because they have C extensions (Pillow, ffi, libevent, etc.).

With Homebrew you at least know to which stable low level libraries it is linked. The big drawback with Homebrew is that they can and will remove your Python3.8 from under your feet with any random update in the future when they decide that Python3.9 or Python3.10 is good enough for their eco system...

-----------

The problem is one of compiling and linking: If the Python executable is  (dy)-linked against MacOSX.sdk from one version (Big Sur) and afterwards you try to install a Python binding to a lower level library that resolves the linking to another MacOSX.sdk version (Monterey), type (Xcode/CLT) or to a similar library in Homebrew instead,  my vague errors start to happen. 

One prime suspect I have now found is libffi, which is required by the Python cffi binding, which is required by zope.security/Btrees/Transaction/etc.  There is an implementation in MacOSX.sdk/CLT (the osx system libffi) and there is a separate library available for Homebrew.  With a combination of ENV variables CFLAGS/LDFLAGS , and  PKG_CONFIG_PATH and pkg-config .pc files that are installed with Homebrew, the compiler/linker picks up the link targets.  

```
> bin/pip install cffi==1.14.5
>  fred$ otool -L ./lib//python3.8/site-packages/_cffi_backend.cpython-38-darwin.so 
./lib//python3.8/site-packages/_cffi_backend.cpython-38-darwin.so:
	/usr/lib/libffi.dylib (compatibility version 1.0.0, current version 27.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.0.0)
```


There is a very long issue thread on the Cryptography github repo where people were investigating and discussing similar issues and amongst the posts is a very insightful one from user Aloysius from feb 13th.  If you install libffi from Homebrew it installs a libffi.pc file that has/had a bug where the compiler hint points to the Homebrew libffi, but the linker hint  points to the system libffi.  :-O  ( https://github.com/pyca/cryptography/issues/5742#issuecomment-778573539 )

I've been setting up multiple python3 & venvs and trying to install a minimal Plone setup with different CFLAGS/LDFLAGS and one of the things I found out I cannot get cffi to bind to the Homebrew ffi if the Python3 I used is from Homebrew as well. 

So that's the ffi part, but There's also my pyenv Python3 linking against the Xcode MacOSX.SDK and the Homebrew prebuilt python linking against Apple CommandLineTools.  Oh joy.  Flashbacks from trying to compile a Linux 1.X kernel in my twenties while studying at university.  

And the unpredictability of all of this mess only get worse when after building a Python Package gets stored in your buildout or pip cache and you run buildout or a pip install that pulls in sub-dependencies (so with my tests from the last 2 weeks I disabled all egg caches). 

Next steps are to recompile a Python 3.8 and/or 3.9 with pyenv, keep everything else stable and see if my theory is correct using otool & other inspection tools. 


```
 fred$ /opt/homebrew/Cellar/python\@3.8/3.8.12_1/bin/python3-config --cflags
-I/opt/homebrew/Cellar/python@3.8/3.8.12_1/Frameworks/Python.framework/Versions/3.8/include/python3.8 -I/opt/homebrew/Cellar/python@3.8/3.8.12_1/Frameworks/Python.framework/Versions/3.8/include/python3.8 -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk

 fred$ ~/.pyenv/versions/3.8.12/bin/python3-config --cflags
-I/Users/fred/.pyenv/versions/3.8.12/include/python3.8 -I/Users/fred/.pyenv/versions/3.8.12/include/python3.8 -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include
```

**Comment by nzambello** [comment]
With the updated packages on homebrew, you can install `python@VERSION` and only that version will be installed.
Python distributed packages under different formulae, so if you install `python@3.8` it won't be involuntarily upgraded anymore.

**Comment by sneridagh** [comment]
@fredvd regarding brew, did you know about this?:

```
# do not update brew on every install
export HOMEBREW_NO_AUTO_UPDATE=1
```

it's essential, otherwise brew tries to update every single formulae, and break all buildouts and environments.

File: api/buildout.cfg [hunk 1]
```diff
@@ -73,14 +72,12 @@ eggs = ${instance:eggs}
 [versions]
 setuptools =
 zc.buildout =
+plone.restapi = 8.14.0
 plone.restapi = 8.15.2
```
Comment (iFlameing) [inline api/buildout.cfg:76]: there should be only one version of restapi

**Comment by iFlameing** [comment]
> > @fredvd did you try again? any other insight?
> > @iFlameing @jackahl Could you please test on your M1s?
> 
> @sneridagh . Yes I have some insights. But i'm not there yet. You can safely merge this PR when it works for others because it's definitely a local edge case on my machine.
> 
> But it's an edge case many more people will run into in the future if they use pyenv.
> 
> I switched back to my good old Intel MB Pro to do the daily work and meanwhile debug/investigate the configuration on my M1 Mac Mini when I had some time left for the last 2 weeks. The main culprit is a combination of me using pyenv for my Python 3 installs, attempting (and succeeding) in compiling python2 using pyenv with Rosetta in 2020 on Big Sur in x86 mode and Apple's Xcode / CommandLineTools (CLT) MacOSX.SDK mess.
> 
> I found this out by testing more after @nzambello mentionned that Python3.8 from Homebrew worked fine for him. So I disabled pyenv and started testing purely with a Homebrew python3.
> 
> TL/DR: Use Homebrew for now, because pyenv can become a minefield of differently picked up compiler/linker paths to different libraries between the Python3 executable and later Python modules that also need to link to the same or other system libraries because they have C extensions (Pillow, ffi, libevent, etc.).
> 
> With Homebrew you at least know to which stable low level libraries it is linked. The big drawback with Homebrew is that they can and will remove your Python3.8 from under your feet with any random update in the future: when they decide that Python3.9 or Python3.10 is good enough for their eco system...
> 
> The problem is one of compiling and linking: If the Python executable is (dy)-linked against MacOSX.sdk from one version (Big Sur) and afterwards you try to install a Python binding to a lower level library that resolves the linking to another MacOSX.sdk version (Monterey), type (Xcode/CLT) or to a similar library in Homebrew instead, the vague errors I saw start to happen.
> 
> One prime suspect I have now found is libffi, which is required by the Python cffi binding, which is required by zope.security/Btrees/Transaction/etc. There is an implementation in MacOSX.sdk, but there are 2 flavors of that one: in the full Xcode.app and one in the 250-400Mb CommandLineTools. CLT (the osx system libffi).
> 
> Aaaand there is a separate library available for Homebrew. With a combination of ENV variables CFLAGS/LDFLAGS , and PKG_CONFIG_PATH and pkg-config .pc files that are installed with Homebrew, the compiler/linker picks up the link targets. Example:
> 
> ```
> > bin/pip install cffi==1.14.5
> >  fred$ otool -L ./lib//python3.8/site-packages/_cffi_backend.cpython-38-darwin.so 
> ./lib//python3.8/site-packages/_cffi_backend.cpython-38-darwin.so:
> 	/usr/lib/libffi.dylib (compatibility version 1.0.0, current version 27.0.0)
> 	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.0.0)
> ```
> 
> There is a very long issue thread on the Cryptography github repo where people were investigating and discussing similar issues and amongst the posts is a very insightful one from user Aloysius from feb 13th. If you install libffi from Homebrew it installs a libffi.pc file that has/had a bug where the compiler hint points to the Homebrew libffi, but the linker hint points to the system libffi. :-O ( [pyca/cryptography#5742 (comment)](https://github.com/pyca/cryptography/issues/5742#issuecomment-778573539) )
> 
> I've been setting up multiple python3 & venvs and trying to install a minimal Plone setup with different CFLAGS/LDFLAGS and one of the things I found out (or my hypothesis is) I cannot get cffi to bind to the Homebrew libffi if the Python3 I used is linked against the ‘wrong’ systemlib.
> 
> So that's the ffi part, but There's also my pyenv Python3 linking against the Xcode MacOSX.SDK and the Homebrew prebuilt python linking against Apple CommandLineTools. Oh joy. Flashbacks from trying to compile a Linux 1.X kernel in my twenties while studying at university.
> 
> And the unpredictability of all of this mess only get worse when after building a Python Package gets stored in your buildout or pip cache and you run buildout or a pip install that pulls in sub-dependencies (so with my tests from the last 2 weeks I disabled all egg caches).
> 
> Next steps are to recompile a Python 3.8 and/or 3.9 with pyenv, keep everything else stable and see if my theory is correct using otool & other inspection tools.
> 
> Also noticeable: homebrew’s Python 3.8 is linked against a Python Framework library collection and my pyenv 3.8.12 not but got the path to the XCode MacOSX.sdk twice on build… :-S
> 
> ```
>  fred$ /opt/homebrew/Cellar/python\@3.8/3.8.12_1/bin/python3-config --cflags
> -I/opt/homebrew/Cellar/python@3.8/3.8.12_1/Frameworks/Python.framework/Versions/3.8/include/python3.8 -I/opt/homebrew/Cellar/python@3.8/3.8.12_1/Frameworks/Python.framework/Versions/3.8/include/python3.8 -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk
> 
>  fred$ ~/.pyenv/versions/3.8.12/bin/python3-config --cflags
> -I/Users/fred/.pyenv/versions/3.8.12/include/python3.8 -I/Users/fred/.pyenv/versions/3.8.12/include/python3.8 -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include
> ```

**Comment by iFlameing** [comment]
sorry!

**Comment by fredvd** [comment]
> Regarding this error in m1. we already solved this by installing a particular version of cffi. The correct version of cffi is 1.14.4 and this is what we are using for our project.

@iFlameing  Changing the pinned version and then seeing it work might in this case not be the direct 'cause -> effect' relation.  Plone 5.2.6 has cffi 1.14.5 pinned and this version works fine according to CI/CD. 

It is possible that you have a 'broken' cffi-1.14.5 in your egg cache or in your pip cache   (run `pip cache list`) that was inserted there by a previous buildout or pip install with the wrong linker information.  Now it sits there and whenever a project needs cffi-1.14.5 it gets pulled from the cache and your build proces tracebacks. when the cffi package is used to build zope.security, Transaction or other python modules. 

When you downgrade to cffi 1.14.4, a new cffi is built, or the previous built version gets pulled from the cache that links to a working libffi. Or what also could explain this is when you build with wheel support a binary wheel is fetched from pypi and that for another version the binary wheel is not (yet) available. 

The pip cache is, just like the buildout egg cache, shared between virtualenv's and projects. But it is enabled by default afaik unlike the buildout egg cache.

**Comment by nileshgulia1** [comment]
> With the updated packages on homebrew, you can install `python@VERSION` and only that version will be installed. Python distributed packages under different formulae, so if you install `python@3.8` it won't be involuntarily upgraded anymore.

Just in case if anyone needs to install python3 in M1 the right way: https://github.com/Homebrew/discussions/discussions/476
